internal/approval.go

```diff
@@ -151,6 +176,59 @@ func computeApprovalStatus(ctx context.Context, c *client, ownerLogin, repoName
 	return status, truncate(description, statusEventDescriptionMaxLength), finalLabels, computeReviewsToRequest(ctx, teams, pendingTeamNames), nil
 }
 
+// return true if there were any changes in the specified directory. If the directory starts with '/' match is done
+// with HasPrefix, otherwise match is done with Contains function
+func isDirectoryChanged(directory string, commitFiles []*github.CommitFile) (bool, error) {
+
+	var startsWith bool
+	if strings.HasPrefix(directory, "/") {
+		startsWith = true
+		directory = strings.TrimPrefix(directory, "/")
+	}
+
+	for _, commitFile := range commitFiles {
+
+		// we are not checking changes (or additions/deletions) because this can be a new or deleted file
+		if commitFile == nil || commitFile.ContentsURL == nil {
+			return false, fmt.Errorf("commit file %+v has nil contents url, skipping", commitFile)
+		}
+
+		relPath, err := contentsUrlToRelPath(*commitFile.ContentsURL)
+		if err != nil {
+			return false, err
+		}
+
+		if startsWith {
+			if strings.HasPrefix(relPath, directory) {
+				return true, nil
+			}
+		} else {
+			if strings.Contains(relPath, directory) {
```

Comment:
can we add a `/` on the end of the directory here(if not exists) so we dont match files like 
`directory = production`

```sh 
this/is/a/file/production_file.tf

```

could also be applied to root level directories?

Reply:
trimming files instead now (updated tests as well)